tfexec/internal/e2etest/providers_schema_test.go

```diff
@@ -25,88 +25,163 @@ func TestProvidersSchema(t *testing.T) {
 	}{
 		{
 			"basic", func(tfv *version.Version) *tfjson.ProviderSchemas {
-				dk := tfjson.SchemaDescriptionKindPlain
-				// HACK: this is a bug? the contstant value is "plaintext" but "plain" is output,
-				// not sure if its in tfjson or what
-				dk = "plain"
-
-				providerName := "registry.terraform.io/hashicorp/null"
+				var providerSchema *tfjson.ProviderSchemas
 
 				if tfv.LessThan(version.Must(version.NewVersion("0.13.0"))) {
```

Comment:
You can reuse the global var `providerAddressMinVersion` here I think since this is really what you want to vary on.

Reply:
I wonder though if would be cleaner to just use a variable key for the map though? are there other parts that vary besides the key?

Reply:
Oh I see thats what it was already. Hard to see what the difference in the branches is here, if it was only those two values, why explode it out?

Reply:
There are several other differences between the two schemas, which made it messy to continue varying them inline. For example, the 0.13 schema has:
```
ConfigSchema: &tfjson.Schema{
  Version: 0,
  Block:   &tfjson.SchemaBlock{},
}
```

while the 0.12 schema has:

```
ConfigSchema: &tfjson.Schema{
  Version: 0,
    Block: &tfjson.SchemaBlock{
    DescriptionKind: "plain",
  },
}
```

Description types are different, and resource-level descriptions are only present from 0.13.